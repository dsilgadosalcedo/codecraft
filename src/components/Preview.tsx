function Preview({ code }: { code: string }) {
	return (
		<iframe
			srcDoc={code}
			className='w-full h-full rounded-xl'
			title='Preview'
		/>
	)
}

export default Preview
