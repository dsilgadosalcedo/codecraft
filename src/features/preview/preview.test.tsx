import { render } from '@testing-library/react'
import React from 'react'
import { describe, expect, it } from 'vitest'

import Preview from './preview'

describe('Preview', () => {
  it('renders an iframe with the given code', () => {
    const code = '<div>Hello World</div>'
    const { container } = render(<Preview code={code} />)
    const iframe = container.querySelector('iframe')
    expect(iframe).toBeInTheDocument()
    expect(iframe).toHaveAttribute('srcDoc', code)
    expect(container).toMatchSnapshot()
  })
})
