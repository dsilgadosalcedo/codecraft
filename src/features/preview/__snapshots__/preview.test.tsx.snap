// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Preview > renders an iframe with the given code 1`] = `
<div>
  <iframe
    class="w-full h-[calc(100vh-64px)] md:h-full rounded-xl"
    srcdoc="<div>Hello World</div>"
    title="Preview"
  />
</div>
`;
